.changeset/foreach-time-travel-results.md

```
@@ -0,0 +1,5 @@
+---
+"@mastra/core": patch
+---
+
+Preserve completed foreach iterations when time traveling after a failed run.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)